core/rule_generator.py

```diff
@@ -816,18 +817,42 @@ def tables(pattern_json: str, rewrite_json: str) -> list:
         patternTables = RuleGenerator.tablesOfASTJson(patternASTJson, [])
         rewriteTables = RuleGenerator.tablesOfASTJson(rewriteASTJson, [])
 
-        # TODO - patternTables should be superset of rewriteTables
-        #
+        # patternTables should be superset of rewriteTables
+        patternSet = defaultdict(list)
+        rewriteSet = defaultdict(list)
+
+        for table in patternTables:
```

Comment:
Please add a comment before this line to explain the meaning of the output patternSet and rewriteSet.

I guess it means for each real table name, put all its aliases as a list under it. Aslo give an example to explain from the input to what output.

We can use the example pattern in `def test_rewrite_rule_remove_self_join():`, which is:
```sql
        SELECT  e1.name, 
                e1.age, 
                e2.salary 
        FROM employee e1, employee e2
        WHERE e1.id = e2.id
        AND e1.age > 17
        AND e2.salary > 35000;
```